pkg/reconciler/pipelinerun/cancel.go

```diff
@@ -99,23 +99,48 @@ func cancelPipelineRun(ctx context.Context, logger *zap.SugaredLogger, pr *v1bet
 func cancelPipelineTaskRuns(ctx context.Context, logger *zap.SugaredLogger, pr *v1beta1.PipelineRun, clientSet clientset.Interface) []string {
 	errs := []string{}
 
-	// Loop over the TaskRuns in the PipelineRun status.
-	// If a TaskRun is not in the status yet we should not cancel it anyways.
-	for taskRunName := range pr.Status.TaskRuns {
-		logger.Infof("cancelling TaskRun %s", taskRunName)
-
-		if _, err := clientSet.TektonV1beta1().TaskRuns(pr.Namespace).Patch(ctx, taskRunName, types.JSONPatchType, cancelTaskRunPatchBytes, metav1.PatchOptions{}, ""); err != nil {
-			errs = append(errs, fmt.Errorf("Failed to patch TaskRun `%s` with cancellation: %s", taskRunName, err).Error())
-			continue
+	// If pr.Status.ChildReferences is populated, use that as source of truth for TaskRun and Run names.
+	if len(pr.Status.ChildReferences) > 0 {
+		// Loop over the ChildReferences in the PipelineRun status.
+		for _, cr := range pr.Status.ChildReferences {
```

Comment:
I might prefer a helper function which returns the runs and taskruns to cancel based on the embedded status value.